### DISCUSSION TITLE
Looking for a node that creates simple horizontal and vertical masks

### DISCUSSION THREAD

**Discussion by andreszs** [WASasquatch/was-node-suite-comfyui#525] [Q&A] [answered]
I'm looking for a node with the ability to create simple masked areas of arbitrary size, with a visual representation of the areas, similar to this **Visual Area Prompt** node:
![image](https://github.com/user-attachments/assets/7f084b45-0e73-4633-8ccc-cfad919340d0)

The only thing I've found so far is the awkward method using the **Simple Binary Pattern** from ComfyRoll Studio:
![image](https://github.com/user-attachments/assets/4f94d654-6207-49a7-a56b-a4911cb64f1d)

However it cannot create vertical masks, in fact it creates only images, then we have to convert it to mask, and I'd prefer to avoid installing a plethora of nodes to use just 1 of them.

If **WAS Node Suite** can create simple area masks, or images that are then converted to mask, kindly let me know which nodes to use, because the documentation has no screenshots and it's hard to mentally parse it all looking for specific nodes like this.

**Comment by WASasquatch** [comment]
I don't think I have anything for direct masks like this, no. I am sure you
could build something with blanks and expansions and using image to mask
and make a combined group node

**Comment by andreszs** [comment]
Using almost no help from GPT, I took the liberty creating a [PR](https://github.com/WASasquatch/was-node-suite-comfyui/pull/526) with two nodes for this.

![image](https://github.com/user-attachments/assets/b3d58804-e3df-443a-8501-4c466be8259f)

I know the nodes are quite basic with no advanced functions but this is the first time I write some python code. Let's hope it's merged because simple rectangular mask nodes with visual preview are nowhere to be found, and this WAS Node Suite is superb.

**Reply by WASasquatch** [reply]
Hi! Sorry, I am late checking things around here. 

I do not allow JavaScript in WAS-NS so it isn't reliant on ComfyUI's UI, and can run in a headless environment for services. Is it possibly to make the activation or your JS optional in the config, and ensure it can function without it (the preview)?

These look great though! Good work for a first try, Id say. Looks like you got a gift in there.

**Reply by andreszs** [reply]
I understand, then I'll soon update this fork to remove the preview canvas and the JS files that won't be required anymore.

**Reply by andreszs** [reply]
✔️ PR #526 has been updated successfully

**Reply by WASasquatch** [reply, marked as answer]
I merged it. Thank you!
